smurfs/src/components/SmurfList.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { connect } from 'react-redux'
+import { getSmurf } from '../actions/actions';
+function SmurfList(props) {
+    console.log("props.smurfs", props.smurfs)
+    const fetchSmurf = e => {
+        e.preventDefault();
+        props.getSmurf();
+    };
+    return (
+        <>
+            <h2>Welcome to SmurfLand!</h2>
+            <div>
+                {props.smurfs && props.smurfs.map(smurf => (
+                    <div key={smurf.id}>
+                        <h4>{smurf.name}</h4>
+                        <p>{smurf.age}</p>
+                        <p>{smurf.height}</p>
+                    </div>
+                )
+                )}
+            </div>
+            <div>
+                {props.error && <p className="error">{props.error}</p>}
+                <button onClick={fetchSmurf}>Fetch Smurf!</button>
```

Comment:
adding fetchSmurf to the onClick here will work fine. But you could save yourself some code if you just did the props.getSmurf function inside the onClick itself, since your fetchSmurf function is merely running the getSmurf function you could cut out the fetchSmurf function all together!